src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/rc_command_request.cc

```diff
@@ -0,0 +1,251 @@
+ SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "rc_rpc_plugin/commands/rc_command_request.h"
+#include "rc_rpc_plugin/rc_module_constants.h"
+#include "application_manager/message_helper.h"
+#include "application_manager/hmi_interfaces.h"
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "RCCommandRequest")
+
+namespace rc_rpc_plugin {
+namespace commands {
+
+RCCommandRequest::RCCommandRequest(
+    rc_rpc_plugin::ResourceAllocationManager& resource_allocation_manager,
+    const app_mngr::commands::MessageSharedPtr& message,
+    app_mngr::ApplicationManager& application_manager)
+    : app_mngr::commands::CommandRequestImpl(message, application_manager)
+    , resource_allocation_manager_(resource_allocation_manager) {}
+
+RCCommandRequest::~RCCommandRequest(){}
+
+bool RCCommandRequest::IsInterfaceAvailable(
+    const app_mngr::HmiInterfaces::InterfaceID interface) const {
+  app_mngr::HmiInterfaces& hmi_interfaces =
+      application_manager_.hmi_interfaces();
+  const app_mngr::HmiInterfaces::InterfaceState state =
+      hmi_interfaces.GetInterfaceState(interface);
+  return app_mngr::HmiInterfaces::STATE_NOT_AVAILABLE != state;
+}
+
+void RCCommandRequest::OnTimeout() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  SetResourceState(
+      (*message_)[strings::msg_params][message_params::kModuleType].asString(),
+      ResourceState::FREE);
+  SendResponse(
+      false, mobile_apis::Result::GENERIC_ERROR, "Request timeout expired");
+}
+
+bool RCCommandRequest::CheckDriverConsent() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  app_mngr::ApplicationSharedPtr app =
+    application_manager_.application(CommandRequestImpl::connection_key());
+  RCAppExtensionPtr extension = resource_allocation_manager_.GetApplicationExtention(app);
+  if (!extension) {
+    return false;
+  }
+  const std::string module_type =
+      (*message_)[strings::msg_params][message_params::kModuleType].asString();
+  rc_rpc_plugin::TypeAccess access = CheckModule(module_type, app);
+
+  if (access == rc_rpc_plugin::kAllowed) {
+    set_auto_allowed(true);
+    return true;
+  } else {
+    SendDisallowed(access);
+  }
+  return false;
+}
+
+rc_rpc_plugin::TypeAccess RCCommandRequest::CheckModule(
+    const std::string& module_type,
+    application_manager::ApplicationSharedPtr app) {
+  return application_manager_.GetPolicyHandler().CheckModule(
+             app->policy_app_id(), module_type)
+             ? rc_rpc_plugin::TypeAccess::kAllowed
+             : rc_rpc_plugin::TypeAccess::kDisallowed;
+}
+
+void RCCommandRequest::SendDisallowed(rc_rpc_plugin::TypeAccess access) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  std::string info;
+  switch (access) {
```

Comment:
@LitvinenkoIra why do we need switch in enum with 2 possible values? 
Can we use if? Or at least make this switch more readable